### PR TITLE
MODCXMUX-30:Add support for codex-packages and codex-packages-sources methods

### PR DESCRIPTION
**Pull request by carolegodfrey**
## Purpose
Per https://issues.folio.org/browse/MODCXMUX-30 we are defining a common RAML definition for codex-packages and codex-packages-sources endpoints.

The following 3 methods are defined:

```
GET /codex-packages
GET /codex-packages/{id}
GET /codex-packages-sources
```
CQL for searching for Codex Packages is simpler than what is supported for Instances, for example

`name=academic and type=Print and ext.selected=true and source=kb sortby name`

Used package field mappings as define at:
https://docs.google.com/spreadsheets/d/1GQhgG5PnXJguB8K_JagmIVYpxx8HwGdkU4Sn96Q6YJg/edit#gid=1804402370

Used package filter information from input defined at:
https://docs.google.com/spreadsheets/d/1GQhgG5PnXJguB8K_JagmIVYpxx8HwGdkU4Sn96Q6YJg/edit#gid=1977344722
https://docs.google.com/spreadsheets/d/1GQhgG5PnXJguB8K_JagmIVYpxx8HwGdkU4Sn96Q6YJg/edit#gid=310763276

## Approach
- Used Codex instances RAML definition as a model for defining packages methods, parameters and object definitions.
- Created a separate definition file for packages methods. These methods will be part of a codex-packages interface that will be implemented by mod-codex-mux, mod-agreements and mod-codex-ekb
- Created codex_package_cqlschema.json defining commonly supported CQL for searching for packages.
- Created codex_package_cqlschema-ext.json defining extension for CQL for searching for selected packages.
- Executed `raml2html codex-packages.raml > codex-packages.html` to generate sample api document
- As test created temporary branch `add-codex-packages` in `mod-codex-ekb` project which references this branch for submodule.

## Screenshots
screen shot 2019-02-12 at 5 03 20 pm
screen shot 2019-02-12 at 5 03 54 pm

## Learning
- https://stackoverflow.com/questions/1777854/how-can-i-specify-a-branch-tag-when-adding-a-git-submodule
- https://dev.folio.org/guides/raml-cop/

### CODE REVIEW

**Review by dcrossleyau — COMMENTED** [review]
I added configuration for this new codex-packages.raml file.

Wow, you added json schema descriptions too. Nice.
There is still the separate job for the other existing schema descriptions (not codex-related) which is why we are still seeing messages from the lint-raml-schema stage of CI.

Please do not re-add that old scripts/lint-raml-cop.sh script. We have the new one configured in CI. See the new one `folio-tools/lint-raml` via the linked doc at https://dev.folio.org/guides/raml-cop/ if needed to run locally.

I have not assessed any other technical aspects of this PR.

**Comment by carolegodfrey** [comment]
> I added configuration for this new codex-packages.raml file.
> 
> Wow, you added json schema descriptions too. Nice.
> There is still the separate job for the other existing schema descriptions (not codex-related) which is why we are still seeing messages from the lint-raml-schema stage of CI.
> 
> Please do not re-add that old scripts/lint-raml-cop.sh script. We have the new one configured in CI. See the new one `folio-tools/lint-raml` via the linked doc at https://dev.folio.org/guides/raml-cop/ if needed to run locally.
> 
> I have not assessed any other technical aspects of this PR.

Thanks for your help @dcrossleyau -- I removed the obsolete script and working through those other missing descriptions! :dizzy_face:

**Comment by dcrossleyau** [comment]
Sorry, i must have confused you. Those other descriptions are some other job (FOLIO-1554), not this one. I was just explaining why Jenkins was reporting about them here.

But it is fantastic that you have done.

File: schemas/codex/packageCollection.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  "properties": {
+    "packages": {
+      "type": "array",
+      "description": "Collection of packages",
+      "packages": {
+        "type": "object",
+        "$ref": "package.json"
+      }
+    },
+    "resultInfo": {
```
Comment (natali-zaitseva) [inline schemas/codex/packageCollection.json:15]: seems that this section should look like 
`"resultInfo": {
  "description": "Result information schema",
  "$ref": "../resultInfo.schema"
    }`
this should fix jenkins  error `ERROR: lint-raml-cop: resultInfo.schema: Missing "description" for: value`

File: schemas/CQLSchema.schema [hunk 2]
```diff
@@ -5,31 +5,37 @@
...
     },
     "prefix": {
-      "type": "string"
+      "type": "string",
+      "description": "Prefix of the property"
     },
     "uri": {
-      "type": "string"
+      "type": "string",
+      "uri": "Uri of the property"
```
Comment (natali-zaitseva) [inline schemas/CQLSchema.schema:22]: to fix jenkins `ERROR: lint-raml-cop: CQLSchema.schema: Missing "description" for: name, type, uri`
we probably should add 
`"description": "Uri of the property"`

File: schemas/CQLSchema.schema [hunk 3]
```diff
@@ -5,31 +5,37 @@
...
           "name": {
             "type": "string"
           },
           "description": {
-            "type": "string"
+            "type": "string",
+            "description": "Description of the Index"
           },
           "type": {
             "type": "string",
```
Comment (natali-zaitseva) [inline schemas/CQLSchema.schema:43]: "description": "Type of the index"

File: schemas/CQLSchema.schema [hunk 4]
```diff
@@ -5,31 +5,37 @@
...
       "description": "If context set is default, the server will determine it automatically"
     },
     "indexes": {
       "type": "array",
+      "description": "Array of CQL Indexes",
       "items": {
         "type": "object",
         "properties": {
           "name": {
             "type": "string"
```
Comment (natali-zaitseva) [inline schemas/CQLSchema.schema]: "description": "Name of the index"

**Comment by carolegodfrey** [comment]
> Sorry, i must have confused you. Those other descriptions are some other job (FOLIO-1554), not this one. I was just explaining why Jenkins was reporting about them here.
> 
> But it is fantastic that you have done.

Just resolved the missing descriptions for all Codex related definitions (both new and existing)

File: schemas/codex/packageCollection.json (hunk 1 above)
Comment (carolegodfrey) [reply schemas/codex/packageCollection.json:15]: Thanks @natali-zaitseva for talking a look and these details -- resolved these!

**Comment by dcrossleyau** [comment]
Hooray. Thanks for your great work.

File: schemas/codex/coverage.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Coverage Schema",
+  "description": "Coverage Schema",
+  "type": "object",
+  "additionalProperties": false,
+  "properties": {
+    "beginCoverage": {
+      "type": "string",
```
Comment (dmtkachenko) [inline schemas/codex/coverage.json:9]: shouldn't we also set 'format' as 'date' for beginCoverage and endCoverage fields
(see https://github.com/joelittlejohn/jsonschema2pojo/wiki/Reference#format)
.. potentially with 'customPattern'

Comment (carolegodfrey) [reply schemas/codex/coverage.json:9]: Thanks for recommendation @dmtkachenko -- added both format and customPattern properties

**Review by dmtkachenko — APPROVED** [review]

**Pull request merged**